pkg/controller/migmigration/stage.go

```diff
@@ -529,6 +529,65 @@ func (t *Task) stagePodReport(client k8sclient.Client) (report PodStartReport, e
 	return
 }
 
+// Match number of stage pods in source and destination cluster
+func (t *Task) allStagePodsMatch() (report []string, err error) {
+	dClient, err := t.getDestinationClient()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	podDList := corev1.PodList{}
+
+	options := k8sclient.MatchingLabels(t.stagePodLabels())
+	err = dClient.List(context.TODO(), options, &podDList)
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	sClient, err := t.getSourceClient()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	podSList := corev1.PodList{}
+	err = sClient.List(context.TODO(), options, &podSList)
+
+	var dPods []string
+
+	for _, pod := range podDList.Items{
+		dPods = append(dPods, pod.Name)
+	}
+
+	for _, pod := range podSList.Items{
+		flag := 0
+		for _, name := range dPods {
+			if name == pod.Name {
+				flag = 1
+				break
+			}
+		}
+		if flag == 0{
+			report = append(report, pod.Name + " is missing. Migration might fail")
+		}
+	}
+	if  len(report) > 0{
+		return
+	}
+
+	stageReport, err := t.ensureDestinationStagePodsStarted()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	if len(stageReport.progress) > 0{
```

Comment:
This will always evaluate to true I think

Reply:
I think checking if !stageReport.started || failed will the be correct check?

Reply:
`progress` field could have content, regardless of the value of the `started` field. So how do I check that all the pods are in running state?